core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -571,7 +572,11 @@ public String toString() {
 
             final BatchResult.Builder<Task> resultBuilder = BatchResult.<Task>builder().successes(tasks);
             if (!electMasterService.hasEnoughMasterNodes(remainingNodesClusterState.nodes())) {
-                return resultBuilder.build(rejoin.apply(remainingNodesClusterState, "not enough master nodes"));
+                final int masterNodes = electMasterService.countMasterNodes(remainingNodesClusterState.nodes());
+                final String message =
+                    LoggerMessageFormat.format("not enough master nodes [{}], but needed [{}]",
```

Comment:
this can be confusing - maybe "not enough master nodes (has [{}], needed [{}])"?